src/Factory/CartItemFactory.php

```diff
@@ -10,45 +10,66 @@
  */
 declare(strict_types=1);
 
-namespace Brille24\SyliusCustomerOptionsPlugin\EventListener;
+namespace Brille24\SyliusCustomerOptionsPlugin\Factory;
 
-use Brille24\SyliusCustomerOptionsPlugin\Entity\OrderItemInterface;
 use Brille24\SyliusCustomerOptionsPlugin\Enumerations\CustomerOptionTypeEnum;
-use Brille24\SyliusCustomerOptionsPlugin\Factory\OrderItemOptionFactoryInterface;
 use Brille24\SyliusCustomerOptionsPlugin\Repository\CustomerOptionRepositoryInterface;
-use Doctrine\ORM\EntityManagerInterface;
-use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+use Sylius\Component\Core\Factory\CartItemFactoryInterface;
 use Sylius\Component\Core\Model\OrderInterface;
-use Sylius\Component\Order\Processor\OrderProcessorInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Product\Resolver\ProductVariantResolverInterface;
+use Sylius\Component\Resource\Factory\FactoryInterface;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\RequestStack;
 use Webmozart\Assert\Assert;
 
-final class AddToCartListener
+class CartItemFactory implements CartItemFactoryInterface
 {
+    private CartItemFactoryInterface $decoratedFactory;
+    private ProductVariantResolverInterface $variantResolver;
+    private RequestStack $requestStack;
+    private OrderItemOptionFactoryInterface $orderItemOptionFactory;
+    private CustomerOptionRepositoryInterface $customerOptionRepository;
+
     public function __construct(
-        private RequestStack $requestStack,
-        private EntityManagerInterface $entityManager,
-        private OrderItemOptionFactoryInterface $orderItemOptionFactory,
-        private OrderProcessorInterface $orderProcessor,
-        private CustomerOptionRepositoryInterface $customerOptionRepository,
-    )
```

Comment:
Is there a reason we are not using constructor property promotion here?

Reply:
Good question ;) Not sure why I went this way in the original PR but we could totally switch to constructor property promotion.

Reply:
That would be great if you could do that.